lib/plek.rb

```diff
@@ -62,24 +73,27 @@ def initialize(domain_to_use = nil, external_domain = nil)
   #   scheme (eg `//foo.example.com`)
   # @return [String] The base URL for the service.
   def find(service, options = {})
-    name = name_for(service)
-    if (service_uri = defined_service_uri_for(name))
+    hostname = clean_name(service)
+    if (service_uri = defined_service_uri_for(hostname))
       return service_uri
     end
 
-    host = "#{name}.#{options[:external] ? external_domain : parent_domain}"
-
-    if (host_prefix = ENV["PLEK_HOSTNAME_PREFIX"])
-      host = "#{host_prefix}#{host}"
+    if should_prefix?(hostname)
+      hostname = "#{host_prefix}#{hostname}"
```

Comment:
Super minor suggestion, so only apply if you're already in the code, but I wonder if this would readable without an extra method.

e.g 

```suggestion
    hostname = "#{host_prefix}#{hostname}" unless unprefixable_hosts.include?(hostname)
```

Reply:
Done.